### ISSUE TITLE
Capacity limit is ignored

### ISSUE THREAD

**Issue by phlegx** [rancher/local-path-provisioner#345]
Hi there!

On the README I read:

```
Cons
No support for the volume capacity limit currently.
The capacity limit will be ignored for now.
```

This means the Local path Provisioner does not respect the limits atm, right?

Are there any plans to implement this and if yes, when could we expect this? This would be a really needed feature. Otherwise one has no control - at least not within Kubernetes - for to avoid potential disk overflow.

best
Martin

**Comment by jcox10** [comment]
This is one of the main reasons we switched to longhorn. Local path really isn't useful beyond testing.

**Comment by phlegx** [comment]
> This is one of the main reasons we switched to longhorn. Local path really isn't useful beyond testing.

Hi @jcox10 ! Thanks for the info! Can Longhorn offer same as Local path provisioner in  regard of local volumes. Meaning creating local volumes on the hard disk of every server?

Ans second question if this holds true: Is there an easy switch from Local Path provisioner volumes to Longhorn ones?

thanks
Martin

**Comment by derekbit** [comment]
> > This is one of the main reasons we switched to longhorn. Local path really isn't useful beyond testing.
> 
> Hi @jcox10 ! Thanks for the info! Can Longhorn offer same as Local path provisioner in regard of local volumes. Meaning creating local volumes on the hard disk of every server?
> 
> Ans second question if this holds true: Is there an easy switch from Local Path provisioner volumes to Longhorn ones?
> 
> thanks Martin

@phlegx 

Can you provide more information on your environment including the network env (1Gbps or 10 Gbps) and the underlying storage (SSD or HDD)?

**Comment by phlegx** [comment]
Hi @derekbit 

Rancher 2.6.8
Provider: RKE1
Kubernetes Version: v1.23.8
SSD NVME on each of our 3 Server  in the Kubernetes Cluster
Network: 1Gbps

**Comment by derekbit** [comment]
> Hi @derekbit
> 
> Rancher 2.6.8 Provider: RKE1 Kubernetes Version: v1.23.8 SSD NVME on each of our 3 Server in the Kubernetes Cluster Network: 1Gbps
> 
> We are curently using lpp to create volumes directly on our SSD disks, which works perfect. Only thing is missing is the limit.

I see. In Longhorn, each volume has one or multiple replicas. The network is only 1Gbps which is a bit slow for Longhorn. However, in your use case, you use a volume with a local replica (set volume dataLocality to `strict-local`), ss the network is not an issue.
You can try **Longhorn v1.4.3** first and see if everything is ok.
https://longhorn.io/docs/1.4.3/

**Comment by ATP3530** [comment]
is there any future plan to support capacity limit?

**Comment by derekbit** [comment]
@ATP3530 @RipperSK 
I don't have time for the capacity limit currently because of the development of Longhorn project.
Any contribution is appreciated.

**Comment by yuvipanda** [comment]
Can this be re-opened? I think this would be a wonderful feature to add. It's listed in the README as a drawback

**Comment by YuryHrytsuk** [comment]
> > Hi [@derekbit](https://github.com/derekbit)
> > Rancher 2.6.8 Provider: RKE1 Kubernetes Version: v1.23.8 SSD NVME on each of our 3 Server in the Kubernetes Cluster Network: 1Gbps
> > We are curently using lpp to create volumes directly on our SSD disks, which works perfect. Only thing is missing is the limit.
> 
> I see. In Longhorn, each volume has one or multiple replicas. The network is only 1Gbps which is a bit slow for Longhorn. However, in your use case, you use a volume with a local replica (set volume dataLocality to `strict-local`), ss the network is not an issue. You can try **Longhorn v1.4.3** first and see if everything is ok. https://longhorn.io/docs/1.4.3/

Will Longhorn `strict-local` volume has exactly the same performance as local Volume (provisioned via `local-path-provisioner`)? Or there is some overhead that will make it slower?

**Comment by derekbit** [comment]
> > > Hi [@derekbit](https://github.com/derekbit)
> > > Rancher 2.6.8 Provider: RKE1 Kubernetes Version: v1.23.8 SSD NVME on each of our 3 Server in the Kubernetes Cluster Network: 1Gbps
> > > We are curently using lpp to create volumes directly on our SSD disks, which works perfect. Only thing is missing is the limit.
> > 
> > 
> > I see. In Longhorn, each volume has one or multiple replicas. The network is only 1Gbps which is a bit slow for Longhorn. However, in your use case, you use a volume with a local replica (set volume dataLocality to `strict-local`), ss the network is not an issue. You can try **Longhorn v1.4.3** first and see if everything is ok. https://longhorn.io/docs/1.4.3/
> 
> Will Longhorn `strict-local` volume has exactly the same performance as local Volume (provisioned via `local-path-provisioner`)? Or there is some overhead that will make it slower?

No. There is extra performance overhead in Longhron v1 strict-local volume.

**Comment by erendogan51** [comment]
support for this feature would a great addition +1
